context.go

```diff
@@ -642,15 +733,31 @@ func (c *Context) WalkBlueprintsFiles(rootFile string, visitor FileVisitor) (dep
 			c.parseOneAsync(blueprint.string, blueprint.Scope, rootDir,
 				errsCh, fileCh, blueprintsCh, depsCh)
 
-			doneCh <- struct{}{}
+			doneCh <- blueprint
 		}()
 	}
 
-	tooManyErrors := false
+	foundParseableBlueprint := func(blueprint stringAndScope) {
+		if activeCount >= maxActiveCount {
+			pending = append(pending, blueprint)
+		} else {
+			startParseBlueprintsFile(blueprint)
+		}
+	}
 
-	startParseBlueprintsFile(stringAndScope{rootFile, nil})
+	startParseDescendants := func(blueprint stringAndScope) {
+		descendants, hasDescendants := descendantsMap[blueprint.string]
+		if hasDescendants {
+			for _, descendant := range descendants {
+				foundParseableBlueprint(stringAndScope{descendant, parser.NewScope(blueprint.Scope)})
+			}
+		}
+	}
 
-	var pending []stringAndScope
+	// begin parsing any files that have no ancestors
+	for _, path := range descendantsMap[""] {
```

Comment:
I'm not sure if its better, but this could be startParseDescendants(stringAndScope{"", parser.NewScope(nil)})

Reply:
ok